### ISSUE TITLE
Stack overflow in output layer forward pass (anything but softmax)

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#540]
If OutputLayer activation function is anything other than softmax: OutputLayer.output(boolean) calls BaseLayer.activate(), which calls OutputLayer.activate(boolean) which calls OutputLayer.output(boolean) -> Stack overflow.
Easiest way to reproduce this is to temporarily modify BackPropMLPTest to use an activation function other than softmax.
I may be able to look into this later today.

**Comment by nyghtowl** [comment]
Fix #541
